### ISSUE TITLE
Trackpad & trackpoint freeze after S3 sleepwake

### ISSUE THREAD

**Issue by paremo29** [kreizlie/Lenovo-Yoga-260-Hackintosh#10]
I'm trying to solve it for weeks.
##### 
I used to install single VoodooPS2 version 2.2.3. Both trackpad and trackpoint are attaching by VoodooPS2 as MagicTrackpad 2. Sometimes trackpoint doesn't work at all once. They also have the issue like title.
##### 
Then I installed VoodooRMI 1.3 + VoodooPS2 2.2.3 with the README (https://github.com/VoodooSMBus/VoodooRMI). The result is the same as single VoodooPS2. VoodooRMI may doesn't function correctly.
##### 
Have you ever faced the problem like this?

**Comment by galaxy-nomad** [comment]
I have the same issue and it also happened with my past HP EliteBook builds with OpenCore. In my case it does not even need to sleep or wake. It just happens running the laptop. The trackpad freezes or becomes unresponsive after a period of time. Very inconvenient if you're a DJ in the middle of rocking the party🤣 I assume there is no fix for this since you had no response and I never received a fix with my older Hackintosh EliteBook builds. I think if you can find a golden build (no issues) that uses Clover that is the best way. OpenCore is imperfect to the point of being unusable in a professional day to day use case.

**Comment by vnln** [comment]
Here is a script I use whenever my trackpad freeze, maybe you can automate it to run on every wake up.

```
#!/bin/bash

sudo kextunload -b de.leo-labs.VoodooSMBus
sleep 2
sudo kextload -b de.leo-labs.VoodooSMBus
```

**Comment by galaxy-nomad** [comment]
Does your trackpad freeze even without sleep wake? Mine freezes just regularly using the laptop while it's awake. I have to do a restart to get the trackpad back. I wonder what causes that kext to malfunction and is there a better kext to use that does not disconnect?

**Comment by vnln** [comment]
Yes, it happened a few times just out of the blue. Now I'm on VoodooSMBus 2.2 & VoodooPS2Controller 2.2.7, working smooth so far.

**Comment by galaxy-nomad** [comment]
Ok great can you make you r EFI available or tell me what I need to do to use your setup. DO I need to delete certain kext files and update my config.plist?

**Comment by vnln** [comment]
Put files like this

```
Kexts Folder
├── VoodooInput.kext (extracted from VoodooPS2Controller.kext)
├── VoodooPS2Controller.kext
│   └── Contents
│       └── PlugIns (delete VoodooPS2Mouse.kext)
│           ├── VoodooPS2Keyboard.kext
│           └── VoodooPS2Trackpad.kext
└── VoodooSMBus.kext
```

and then just edit your config.plist using ProperTree: Cmd+R and Cmd+S

**Comment by cyberfunk** [comment]
Do I need to clear kext cache or can I just do a restart?

**Comment by cyberfunk** [comment]
Thanks so much for your help and suggestions. I've made the changes and will hope that this solves the trackpad issue:-)

**Comment by cyberfunk** [comment]
I just had my first trackpad freeze. That didn't take long at all🤣  Now here's the interesting thing: it may have been a coincidence but everything was working fine until I touched the red touchpoint button and then the trackpad froze/became disconnected. Like I said it may have just been perfect timing but I have now disabled the touchpoint in my BIOS and I'm going to see how long I can go without a freeze.

**Comment by cyberfunk** [comment]
Is VoodooInput.kext the specific kext that enables the red track-point button? I think that red button is the culprit for the freezing disconnecting touchpad. There must be a conflict somewhere. I've been using my Laptop all day and have not yet had a freeze/disconnect with the red track-point button disabled in the BIOS. If VoodooInput.kext controls the red button I would like to delete it altogether if possible.

**Comment by galaxy-nomad** [comment]
Hi vnln. I think I have solved the issue. You need to disable the track-point red button in your BIOS and the freezing trackpad issues should go away for good. I and another user have been testing my theory for a few days and have had absolutely zero freezing issues since we disabled the track-point.

**Comment by timothy-ha** [comment]
As a trackpoint user that is disappointing to hear.  I’d rather have the trackpoint over the trackpad, but even disabling the trackpad in bios does not prevent it from being detected in the OS.
